src/megatron/bridge/training/config.py

```
@@ -1667,6 +1667,20 @@ def _calculate_scheduler_steps(self) -> None:
             else:
                 self.scheduler.lr_warmup_steps = self.scheduler.lr_warmup_iters * self.train.global_batch_size
 
+        # Enforce the Megatron Core invariant: lr_warmup_steps must be < lr_decay_steps.
+        # This can be violated when train_iters is small (e.g. smoke / tutorial runs) while
+        # lr_warmup_iters is tuned for a full-length training run.
+        if self.scheduler.lr_warmup_steps >= self.scheduler.lr_decay_steps:
+            capped = max(0, self.scheduler.lr_decay_steps - 1)
+            warnings.warn(
```

Flake8 (7.3.0)

[error] undefined name 'warnings'

(F821)